lib/Agrammon/OutputFormatter/Excel.pm6

```diff
@@ -70,38 +68,45 @@ sub output-as-excel(
                         my $value = flat-value($raw-value // 'UNDEFINED');
                         my $var-print = $model.output-print($module, $output) ~ ',All';
                         if not $prints or $var-print.split(',') ∩ @print-set {
-                            $n++;
-                            my $unit = $model.output-unit($module, $output, $language);
+                            my $order        = $model.output-labels($module, $output)<sort> || 'no order';
+                            my $unit         = $model.output-unit($module, $output, $language);
                             my $output-label = $language ?? $model.output-labels($module, $output){$language} !! $output;
                             my $unit-label   = $language ?? $model.output-units($module, $output){$language}  !! $unit;
-
-                            if $module ne $last-module {
-                                $output-sheet.set($row, $col + 0, $module);
-                                $last-module = $module;
-                            }
-                            $output-sheet.set($row, $col+1, $q-name);
-                            $output-sheet.set($row, $col+2, $output-label // 'Output: ???');
-                            $output-sheet.set($row, $col+3, $value, :number-format('#,###'));
-                            $output-sheet.set($row, $col+4, $unit-label // 'Unit: ???');
-                            $row++;
+                            @records.push( %( :module($q-name), :$output-label, :$value, :$unit-label, :$order ) );
                             if $include-filters {
                                 if $raw-value ~~ Agrammon::Outputs::FilterGroupCollection && $raw-value.has-filters {
-                                    render-filters($output-sheet, $row, $col, $model, $raw-value, $unit, $language, :$all-filters);
+                                    push-filters(@records, $q-name, $model, $raw-value, $unit-label, $language, $order, :$all-filters);
                                 }
                             }
                         }
                     }
                 }
             }
         }
-        NEXT $row++ if $n; # empty line between sections
     }
 
+    my $col = 0;
+    my $last-module = '';
+    for @records.sort(+*.<order>) -> %rec {
+        %rec<module> ~~ / (.+?) '::' /;
+        my $module = $0 // %rec<module>;
+        if $module ne $last-module {
+            $output-sheet.set($row, $col+0, ~$module, :bold);
+            $last-module = $module;
+            $row++;
+        }
+        $output-sheet.set($row, $col+1, %rec<module>);
+        $output-sheet.set($row, $col+2, %rec<output-label> // 'Output: ???');
+        $output-sheet.set($row, $col+3, %rec<value>, :number-format('#,###'));
+        $output-sheet.set($row, $col+4, %rec<unit-label> // 'Unit: ???');
+        $output-sheet.set($row, $col+5, %rec<order>);
```

Comment:
Hm, curious that we output the sort order

Reply:
Just for debugging (also for Christoph, who needs to tune that.